bluesky/protocols.py

```diff
@@ -134,6 +137,10 @@ def add_callback(self, callback: Callable[["Status"], None]) -> None:
         """
         ...
 
+    @property
+    def exception(self) -> Optional[StatusException]:
+        ...
+
```

Comment:
Why should it raise? It could just be an unused parameter, no? Or would this cause an issue with flake8/isort/linting?

Reply:
It should raise because it can't honour a timeout of anything other than 0, asyncio does not allow you to block in a sync function while still keeping the event loop alive. If the user requests anything other than 0, then they are explicitly asking for a wait with timeout which we can't do, so it should raise saying so.